### ISSUE TITLE
how can i add custom attributes when signup

### ISSUE THREAD

**Issue by zhangli-jerry** [jonsaw/amazon-cognito-identity-dart#68]
how can i add custom attributes when signup?
can i use ClientMataData ?

Thanks.

**Comment by shierro** [comment]
seems so
https://github.com/jonsaw/amazon-cognito-identity-dart/blob/676687346a963b37c96d7d52333d06153d9f69bb/lib/src/cognito_user.dart#L408

1st time we've tried, we got error code 400 SerializationException

**Comment by furaiev** [comment]
Guys, fixed in forked branch https://pub.dev/packages/amazon_cognito_identity_dart_2
Welcome to contribute

**Comment by shierro** [comment]
@furaiev thanks for fixing this one quickly. i think it's also good to submit a PR to this repo so we can switch back to it once it is merged

**Comment by furaiev** [comment]
I don't believe that owner will start support this package again. There are already 14 unresolved PRs.

**Comment by Aathi** [comment]
```dart
final userAttributes = [
      new AttributeArg(name: 'name', value: name),
      new AttributeArg(name: 'custom:attribute_1', value: attribute_1),
      new AttributeArg(name: 'custom:attribute_2', value: attribute_2),
      new AttributeArg(name: 'custom:attribute_3', value: attribute_3),
]
```
